src/containers/forms/PlanForm/helpers.ts

```diff
@@ -0,0 +1,339 @@
+  PlanGoal,
+  PlanGoalDetail,
+  PlanGoaldetailQuantity,
+  PlanGoalTarget,
+  UseContext,
+} from '../../../configs/settings';
+import { DATE, IRS_TITLE, IS, NAME, REQUIRED } from '../../../constants';
+import { InterventionType, PlanStatus } from '../../../store/ducks/plans';
+
+/** separate FI and IRS activities */
+export const FIActivities = omit(planActivities, ['IRS']);
+export const IRSActivities = pick(planActivities, ['IRS']);
+
+/** Array of FI Statuses */
+export const fiStatusCodes = Object.values(FIClassifications).map(e => e.code as FIStatusType);
+
+/** Yup validation schema for PlanForm */
+export const PlanSchema = Yup.object().shape({
+  activities: Yup.array().of(
+    Yup.object().shape({
+      actionCode: Yup.string().oneOf(PlanActionCodes.map(e => e)),
+      actionDescription: Yup.string().required(REQUIRED),
+      actionReason: Yup.string()
+        .oneOf(Object.values(actionReasons))
+        .required(REQUIRED),
+      actionTitle: Yup.string().required(REQUIRED),
+      goalDescription: Yup.string().required(REQUIRED),
+      goalDue: Yup.date().required(REQUIRED),
+      goalPriority: Yup.string()
+        .oneOf(Object.values(goalPriorities))
+        .required(REQUIRED),
+      goalValue: Yup.number()
+        .min(1)
+        .required(REQUIRED),
+      timingPeriodEnd: Yup.date().required(REQUIRED),
+      timingPeriodStart: Yup.date().required(REQUIRED),
+    })
+  ),
+  caseNum: Yup.string(),
+  date: Yup.string().required(`${DATE} ${IS} ${REQUIRED}`),
+  end: Yup.date().required(REQUIRED),
+  fiReason: Yup.string().oneOf(FIReasons.map(e => e)),
+  fiStatus: Yup.string().oneOf(fiStatusCodes),
+  interventionType: Yup.string()
+    .oneOf(Object.values(InterventionType))
+    .required(REQUIRED),
+  name: Yup.string().required(`${NAME} ${IS} ${REQUIRED}`),
+  opensrpEventId: Yup.string(),
+  start: Yup.date().required(REQUIRED),
+  status: Yup.string()
+    .oneOf(Object.values(PlanStatus))
+    .required(REQUIRED),
+  title: Yup.string().required(REQUIRED),
+});
+
+/** Plan activity form fields interface */
+export interface PlanActivityFormFields {
+  actionCode: string;
+  actionDescription: string;
+  actionReason: string;
+  actionTitle: string;
+  goalDescription: string;
+  goalDue: Date;
+  goalPriority: string;
+  goalValue: number;
+  timingPeriodEnd: Date;
+  timingPeriodStart: Date;
+}
+
+/** Plan form fields interface */
+export interface PlanFormFields {
+  activities: PlanActivityFormFields[];
+  caseNum?: string;
+  date: Date;
+  end: Date;
+  fiReason?: FIReasonType;
+  fiStatus?: FIStatusType;
+  interventionType: InterventionType;
+  name: string;
+  opensrpEventId?: string;
+  start: Date;
+  status: PlanStatus;
+  title: string;
+}
+
+/**
+ * Convert a plan activity object to an object that can be used in the PlanForm
+ * activities section
+ * @param activityObj - the plan activity object
+ */
+export function extractActivityForForm(activityObj: PlanActivity): PlanActivityFormFields {
```

Comment:
Do you think there is a better place for these types of extractor functions? In this case here probably makes sense b/c the extractor is so specific to the needs of the Form. For more general extractors should they live in the ducks or should there be a similar `helpers` for ducks? (no need to decide before the release, just something to keep in mind!)

Reply:
I dont know where to best put them.  For now since they are only used for PlanForm we have them here, but I think you are right about helpers for ducks.  We should do that.

Reply:
Cool, created an issue for this https://github.com/onaio/reveal-frontend/issues/252